src/features/member/sagas/promoteDelegatedAccount.saga.ts

```diff
@@ -1,14 +1,42 @@
 import { MemberAction } from '$/features/member'
+import { TokenGatedRoomAction } from '$/features/tokenGatedRooms'
+import { Address, EnhancedStream, PrivacySetting } from '$/types'
 import handleError from '$/utils/handleError'
 import setMultiplePermissions from '$/utils/setMultiplePermissions'
 import takeEveryUnique from '$/utils/takeEveryUnique'
 import { error, success } from '$/utils/toaster'
+import { put } from 'redux-saga/effects'
 import { StreamPermission } from 'streamr-client'
 
 function* onPromoteDelegatedAccountAction({
-    payload: { roomId, delegatedAddress, provider, requester, streamrClient },
+    payload: { roomId, delegatedAddress, provider, requester, streamrClient, privacy },
 }: ReturnType<typeof MemberAction.promoteDelegatedAccount>) {
     try {
+        // split for token-gated rooms
+        if (privacy === PrivacySetting.TokenGated) {
```

Comment:
Should we go with 2 different actions, 1 for token-gated rooms and another one for regular rooms? Feels like it'd add clarity.

PS. Naming is hard. It'd have to be something like `onTokenGatedPromoteDelegatedAccountAction`. 🤔

Reply:
Sounds like a sensitive option, i've left the split inside the `usePromoteDelegatedAccount` hook